easy-connect.cpp

```diff
@@ -0,0 +1,277 @@
+    }
+#  endif
+    connect_success = cellular.connect();
+    network_interface = &cellular;
+#elif MBED_CONF_APP_NETWORK_INTERFACE == ETHERNET
+    if (log_messages) {
+        printf("[EasyConnect] Using Ethernet\n");
+    }
+    network_interface = &eth;
+    connect_success = eth.connect();
+#endif
+
+#ifdef EASY_CONNECT_MESH
+    if (log_messages) {
+        printf("[EasyConnect] Using Mesh (%s)\n", EASY_CONNECT_MESH_TYPE);
+        printf("[EasyConnect] Connecting to Mesh...\n");
+    }
+    network_interface = &mesh;
+    mesh.initialize(&rf_phy);
+    connect_success = mesh.connect();
+#endif
+    if(connect_success == 0) {
+        if (log_messages) {
+            printf("[EasyConnect] Connected to Network successfully\n");
+            print_MAC(network_interface, log_messages);
+        }
+    } else {
+        if (log_messages) {
+            print_MAC(network_interface, log_messages);
+            printf("[EasyConnect] Connection to Network Failed %d!\n", connect_success);
+        }
+        return NULL;
+    }
+    const char *ip_addr  = network_interface->get_ip_address();
+    if (ip_addr == NULL) {
+        if (log_messages) {
+            printf("[EasyConnect] ERROR - No IP address\n");
+        }
+        return NULL;
+    }
+
+    if (log_messages) {
+        printf("[EasyConnect] IP address %s\n", ip_addr);
+    }
+    return network_interface;
+}
+
+/* \brief easy_connect - easy_connect function to connect the pre-defined network bearer,
+ *                       config done via mbed_app.json (see README.md for details).
+ *                       This version is just a helper version and uses the easy_connect() with
+ *                       one parameters to do it's job.
+ * IN: bool  log_messages  print out diagnostics or not.
+ *     char* WiFiSSID      WiFi SSID - pointer to WiFi SSID, but if it is NULL  
+ *                         then MBED_CONF_APP_WIFI_SSID will be used
+ *     char* WiFiPassword  WiFi Password - pointer to WiFI password, but if it's NULL
+ *                         then MBED_CONF_APP_WIFI_PASSWORD will be used
+ */
+
+NetworkInterface* easy_connect(bool log_messages,
+                               char* WiFiSSID,
+                               char* WiFiPassword ) {
+
+// This functionality only makes sense when using WiFi
+#if defined (EASY_CONNECT_WIFI) 
+    // We essentially want to populate the _ssid and _password and then call easy_connect() again. 
+    if (WiFiSSID != NULL) {
+        if(strlen(WiFiSSID) > WIFI_SSID_MAX_LEN) {
+            printf("WARNING - WiFi SSID is too long - it will be cut to %d chars.\n", WIFI_SSID_MAX_LEN);
+            return 0;
+        }
+        _ssid = WiFiSSID;
+    }
+
+    if (WiFiPassword != NULL) {
+        if(strlen(WiFiPassword) > WIFI_PASSWORD_MAX_LEN) {
+            printf("WARNING - WiFi Password is too long - it will be cut to %d chars.\n", WIFI_PASSWORD_MAX_LEN);
+            return 0;
```

Comment:
This message is no longer true now that the `return 0` is added.

I think it'd be an ok trade off to use dynamic allocation in this case to remove this failure mode. The amount memory should be very small and it just removes this edge case which would only be found at run time.

Reply:
Ah ok, I didn't even think to look for that! I guess that's not bad then.

I'll keep an eye out for your update on the message 👍

Reply:
It's fixed now.